starlite/datastructures/headers.py

```diff
@@ -41,18 +46,20 @@ class Headers(CIMultiDictProxy[str], MultiMixin[str]):
             libs.org/en/stable/multidict.html#cimultidictproxy).
     """
 
-    def __init__(self, headers: Optional[Union[Mapping[str, str], "RawHeadersList", MultiMapping]] = None) -> None:
+    def __init__(self, headers: Optional[Union[Mapping[str, str], "RawHeaders", MultiMapping]] = None) -> None:
         """Initialize `Headers`.
 
         Args:
             headers: Initial value.
         """
         if not isinstance(headers, MultiMapping):
             headers_: Union[Mapping[str, str], List[Tuple[str, str]]] = {}
-            if isinstance(headers, list):
-                headers_ = [(key.decode("latin-1"), value.decode("latin-1")) for key, value in headers]
-            elif headers:
-                headers_ = headers
+            if headers:
+                if isinstance(headers, Mapping):
+                    headers_ = headers  # pyright: ignore
+                else:
+                    headers_ = [(key.decode("latin-1"), value.decode("latin-1")) for key, value in headers]
+
```

Comment:
I think this whole section should be simplified to

```python
        if headers and not isinstance(headers, MultiMapping):
            headers_: Union[Mapping[str, str], List[Tuple[str, str]]] = {}
            if isinstance(headers, Mapping):
                headers_ = headers  # pyright: ignore
            else:
                headers_ = [(key.decode("latin-1"), value.decode("latin-1")) for key, value in headers]

            super().__init__(CIMultiDict(headers_))
        else:
            super().__init__(headers)
```

Reply:
@provinzkraut uhm, not sure that we can do this replacement. If `headers` is None in your code we will invoke `super().__init__(None)` while the original code will perform `super().__init__(CIMultiDict({}))` in this case.

I might be wrong though as I am writing from my phone

Reply:
`headers and not isinstance(headers, MultiMapping)` is `False` when `headers` is `None`, which means it will go directly to `super().__init__(headers)` (which can handle `None`).

Reply:
No, it can not handle `None`. Try it yourself

```
> python
...
>>> from starlite.datastructures.headers import Headers
>>> Headers(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jtraub/dev/starlite/starlite/datastructures/headers.py", line 65, in __init__
    super().__init__(headers)
TypeError: ctor requires CIMultiDict or CIMultiDictProxy instance, not <class 'NoneType'>
```

Reply:
Oh. Yea, I guess that makes sense.. :upside_down_face: